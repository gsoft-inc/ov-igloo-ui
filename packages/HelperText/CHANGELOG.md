# @igloo-ui/message

## 0.2.10

### Patch Changes

- c3efaf5: Fixed tests and snapshots since react-aria generated ids changed and viewBox was added to icons.

## 0.2.9

### Patch Changes

- a41e1d4: Fix: moved destructuring of props to function parameters so that storybook will automatically show the default value in the props table. It was done in Modal because it needed to send props to useDialog. In this case, @default annotation was used in comments.

## 0.2.8

### Patch Changes

- 6ee9c45: Updated anything causing errors or warnings in the browser console

## 0.2.7

### Patch Changes

- 0640409: Update @igloo-ui/icons to 1.6.0
- 0640409: udpate classnames to 2.3.2

## 0.2.6

### Patch Changes

- d971143: All tests were updated to use testing library and not enzyme

## 0.2.5

### Patch Changes

- 1a3322f: Upgrade igloo icons package from `1.1.0` to `1.3.0`

## 0.2.4

### Patch Changes

- 38a4994: Update dependency on @igloo-ui/icons and other @igloo libs

## 0.2.3

### Patch Changes

- 0a86f13: Fix color of error appearance
- 0a86f13: Update content of README

## 0.2.2

### Patch Changes

- 8c8686e: Change @import of font.css file for a @use in SASS files

## 0.2.1

### Patch Changes

- ba109dd: add props for passing className to HelperText

## 0.2.0

### Minor Changes

- e8f201c: Rename component `Message` by `HelperText`

## 0.1.0

### Minor Changes

- cf5b505: Initial release
