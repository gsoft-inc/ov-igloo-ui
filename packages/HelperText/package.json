{
  "name": "@igloo-ui/helper-text",
  "version": "0.2.0",
  "main": "dist/HelperText.js",
  "module": "dist/HelperText.js",
  "types": "dist/HelperText.d.ts",
  "styles": "dist/helper-text.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Franck franck.gaudin@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "dependencies": {
    "@igloo-ui/icons": "^0.6.6",
    "@igloo-ui/tokens": "^1.0.0",
    "classnames": "^2.3.1"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
