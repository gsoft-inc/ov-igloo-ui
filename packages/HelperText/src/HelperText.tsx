import * as React from "react";
import cx from "classnames";
import Alert from "@igloo-ui/icons/dist/Alert";

import "./helper-text.scss";

export interface HelperTextProps extends React.ComponentProps<"div"> {
    /** Add a specific class to the `HelperText` */
    className?: string;
    /** The content to display inside the HelperText */
    children: React.ReactNode;
    /** Changes the appearance of component */
    error?: boolean;
}

const HelperText: React.FunctionComponent<HelperTextProps> = ({
        children,
        error,
        className,
        ...rest
    }: HelperTextProps) => {
        const classes = cx("ids-helperText", className, {
            [`ids-helperText--error`]: error
        });

        return (
            <span className={classes} {...rest}>
          {error && <Alert className="ids-helperText__icon" size="small" />}
                {children}
        </span>
    );
};

export default HelperText;
