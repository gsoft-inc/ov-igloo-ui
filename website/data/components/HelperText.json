[
  [
    {
      "tags": {},
      "filePath": "/Users/franck.gaudin/Devel/Igloo/components/packages/HelperText/src/HelperText.tsx",
      "description": "",
      "displayName": "HelperText",
      "methods": [],
      "props": {
        "className": {
          "defaultValue": null,
          "description": "Add a specific class to the `HelperText`",
          "name": "className",
          "parent": {
            "fileName": "components/packages/HelperText/src/HelperText.tsx",
            "name": "HelperTextProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/HelperText/src/HelperText.tsx",
              "name": "HelperTextProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "children": {
          "defaultValue": null,
          "description": "The content to display inside the HelperText",
          "name": "children",
          "parent": {
            "fileName": "components/packages/HelperText/src/HelperText.tsx",
            "name": "HelperTextProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/HelperText/src/HelperText.tsx",
              "name": "HelperTextProps"
            },
            {
              "fileName": "components/node_modules/@types/react/index.d.ts",
              "name": "TypeLiteral"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "error": {
          "defaultValue": null,
          "description": "Changes the appearance of component",
          "name": "error",
          "parent": {
            "fileName": "components/packages/HelperText/src/HelperText.tsx",
            "name": "HelperTextProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/HelperText/src/HelperText.tsx",
              "name": "HelperTextProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        }
      }
    }
  ]
]
