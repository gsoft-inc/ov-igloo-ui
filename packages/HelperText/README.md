# HelperText

Providing additional information about a form input.

<Example />

<ReferenceLinks />

## Installation

To install `@igloo-ui/helper-text` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/helper-text
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/helper-text
```

## Usage

Then to use the component in your code just import it!

```jsx
import HelperText from '@igloo-ui/helper-text';

<HelperText>Additional information</HelperText>;
```
