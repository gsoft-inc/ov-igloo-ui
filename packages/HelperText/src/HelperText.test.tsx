/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';

import HelperText from './HelperText';

const childText = 'Information message';

const setUp = (props = {}) => {
  return render(<HelperText {...props}>{childText}</HelperText>);
};

describe('HelperText', () => {
  test('It should render without errors', () => {
    setUp();
    const wrapper = screen.getByText(childText);
    expect(wrapper).toBeInTheDocument();
  });

  test('It should render a default appearance', () => {
    const { asFragment } = setUp();
    expect(asFragment()).toMatchSnapshot();
  });

  test('It should render a error appearance', () => {
    const { asFragment } = setUp({ appearance: 'error' });
    expect(asFragment()).toMatchSnapshot();
  });
});
