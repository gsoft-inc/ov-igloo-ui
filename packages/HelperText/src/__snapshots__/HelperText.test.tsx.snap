// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelperText It should render a default appearance 1`] = `
<DocumentFragment>
  <span
    class="ids-helperText"
  >
    Information message
  </span>
</DocumentFragment>
`;

exports[`HelperText It should render a error appearance 1`] = `
<DocumentFragment>
  <span
    appearance="error"
    class="ids-helperText"
  >
    Information message
  </span>
</DocumentFragment>
`;
